### PR TITLE
Adding support for new time partition periods

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

File: connector/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java [hunk 1]
```diff
@@ -88,7 +87,7 @@
   com.google.common.base.Optional<String> partitionField = empty();
   Long partitionExpirationMs = null;
   com.google.common.base.Optional<Boolean> partitionRequireFilter = empty();
-  com.google.common.base.Optional<String> partitionType = empty();
+  com.google.common.base.Optional<TimePartitioning.Type> partitionType = empty();
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java:90]: Is TimePartitioning the only type of partitioning we will use now?

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryPartitionedTablesEndToEndITSuite.scala [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    testDataset = s"spark_bigquery_${getClass.getSimpleName}_${System.currentTimeMillis()}"
+    IntegrationTestUtils.createDataset(testDataset)
+  }
+
+  override def afterAll: Unit = {
+    IntegrationTestUtils.deleteDatasetAndTables(testDataset)
+    spark.stop()
+  }
+
+
```
Comment (Gaurangi94) [inline connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryPartitionedTablesEndToEndITSuite.scala:81]: nit: unnecessary empty lines

File: connector/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java (hunk 1 above)
Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java:90]: Yes.

File: connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala [hunk 3]
```diff
@@ -151,15 +150,32 @@ case class BigQueryWriteHelper(bigQuery: BigQuery,
...
+    lazy val finishedJob = job.waitFor()
+    try {
+      if (finishedJob.getStatus.getError != null) {
+        throw new BigQueryException(
+          BaseHttpServiceException.UNKNOWN_CODE,
+          s"""Failed to load to ${friendlyTableName} in job ${job.getJobId}. BigQuery error was
+             |${finishedJob.getStatus.getError.getMessage}""".stripMargin.replace('\n', ' '),
+          finishedJob.getStatus.getError)
+      } else {
+        logInfo(s"Done loading to ${friendlyTableName}. jobId: ${job.getJobId}")
```
Comment (Gaurangi94) [inline connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala:162]: nit: else block is not needed here

File: connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala [hunk 4]
```diff
@@ -151,15 +150,32 @@ case class BigQueryWriteHelper(bigQuery: BigQuery,
...
+      }
+    } catch {
+      case e: Exception =>
+        val partitionType = options.getPartitionTypeOrDefault()
+
+        if (e.getMessage.equals(s"Cannot output $partitionType partitioned data in LegacySQL")
```
Comment (Gaurangi94) [inline connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala:168]: can we have a test for this?

**Review by Gaurangi94 — APPROVED** [review]

File: connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala (hunk 4 above)
Comment (davidrabinowitz) [reply connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala:168]: It has been tested against the API

**Pull request merged**